### PR TITLE
Contribution doc

### PR DESCRIPTION
**Pull request by hetunandu**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+Feel free to propose changes to this document in a pull request.
+
+### Table of contents
+- [Code of conduct](CODE_OF_CONDUCT.md)
+- [How can I contribute?](#how-can-i-contribute)
+- [How to set up local development?](#how-to-set-up-local-development)
+- [How to run tests?](#how-to-run-tests)
+
+## How can I contribute?
+There are many places ways can contribute to Appsmith and all types of contributions are highly appreciated.
```
Comment (riodeuno) [inline CONTRIBUTING.md:14]: There are many ways in which we/one can to contribute to Appsmith.  All contributions are highly appreciated.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,122 @@
+# Contributing to Appsmith
+
+First off, thanks for your interest in Appsmith and taking the time to contribute on this project. 🙌
```
Comment (riodeuno) [inline CONTRIBUTING.md:3]: Thank you for your interest..

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+- Documentation
+- Code contribution
+    - Widgets
+    - Datasources
+
+
+## How to set up local development
+
+### Client
+Our frontend client is written in ReactJS and Typescript. State management is done via Redux and workflows
```
Comment (riodeuno) [inline CONTRIBUTING.md:28]: Appsmith's client (UI/frontend) uses the ReactJS library and Typescript. The application also uses libraries like react-redux and redux-saga..

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+
+
+## How to set up local development
+
+### Client
+Our frontend client is written in ReactJS and Typescript. State management is done via Redux and workflows 
+are handled via Redux Saga.
+ 
+##### Pre-requisites:
+
```
Comment (riodeuno) [inline CONTRIBUTING.md:32]: On the development/local machine, please make sure that:

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+1. You have `docker` installed in your system. If not, please visit: [https://docs.docker.com/get-docker/](https://docs.docker.com/get-docker/)
+2. You have `mkcert` installed. Please visit: [https://github.com/FiloSottile/mkcert#installation](https://github.com/FiloSottile/mkcert#installation) for details.
```
Comment (riodeuno) [inline CONTRIBUTING.md:34]: (For `mkcert` to work with firefox `nss` may be required. Details are in the link)

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+3. Run `yarn build` (optional)
+4. Run `yarn start` if you are hitting [https://release-api.appsmith.com](https://release-api.appsmith.com) (Staging environment) as your API backend.
+5. If you are hitting any other API endpoint, 
+    - Please run:
+
+    ```shell script
+    REACT_APP_ENVIRONMENT=DEVELOPMENT HOST=dev.appsmith.com craco start
+    ```
+
+    - Change the API endpoint in the Nginx configuration available in `docker/templates/nginx-linux.conf.template` or `docker/templates/nginx-mac.conf.template`
```
Comment (riodeuno) [inline CONTRIBUTING.md:74]: We will have to run `start-https.sh` again.

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+
+6. Go to [https://dev.appsmith.com](https://dev.appsmith.com) on your browser
+
+#### If you are unable to run docker:
+
+1. Make the values in `nginx-mac.conf.template` empty. None of those properties are required.
```
Comment (riodeuno) [inline CONTRIBUTING.md:80]: @hetunandu  Do you think the users will need a screencast for this troubleshooting. A youtube link maybe?

**Review by riodeuno — CHANGES_REQUESTED** [review]

**Review by mohanarpit — APPROVED** [review]

**Pull request merged**